src/runmode-af-packet.c

```diff
@@ -355,6 +366,113 @@ static void *ParseAFPConfig(const char *iface)
         }
     }
 
+    if (ConfGetChildValueWithDefault(if_root, if_default, "ebpf-lb-file", &ebpf_file) != 1) {
+        aconf->ebpf_lb_file = NULL;
+    } else {
+        SCLogInfo("af-packet will use '%s' as eBPF load balancing file",
+                  ebpf_file);
+        aconf->ebpf_lb_file = ebpf_file;
+    }
+
+#ifdef HAVE_PACKET_EBPF
+    /* One shot loading of the eBPF file */
+    if (aconf->ebpf_lb_file && cluster_type == PACKET_FANOUT_EBPF) {
+        int ret = EBPFLoadFile(aconf->ebpf_lb_file, "loadbalancer",
+                               &aconf->ebpf_lb_fd, EBPF_SOCKET_FILTER);
+        if (ret != 0) {
+            SCLogError(SC_ERR_INVALID_VALUE, "Error when loading eBPF lb file");
+        }
+    }
+#else
+    if (aconf->ebpf_lb_file) {
+        SCLogError(SC_ERR_UNIMPLEMENTED, "eBPF support is not build-in");
+    }
+#endif
+
+    if (ConfGetChildValueWithDefault(if_root, if_default, "ebpf-filter-file", &ebpf_file) != 1) {
+        aconf->ebpf_filter_file = NULL;
+    } else {
+        SCLogInfo("af-packet will use '%s' as eBPF filter file",
+                  ebpf_file);
+        aconf->ebpf_filter_file = ebpf_file;
+        ConfGetChildValueBoolWithDefault(if_root, if_default, "bypass", &conf_val);
+        if (conf_val) {
+            SCLogConfig("Using bypass kernel functionality for AF_PACKET (iface %s)",
+                    aconf->iface);
+            aconf->flags |= AFP_BYPASS;
+        }
+    }
+
+    /* One shot loading of the eBPF file */
+    if (aconf->ebpf_filter_file) {
+#ifdef HAVE_PACKET_EBPF
+        int ret = EBPFLoadFile(aconf->ebpf_filter_file, "filter",
+                               &aconf->ebpf_filter_fd, EBPF_SOCKET_FILTER);
+        if (ret != 0) {
+            SCLogError(SC_ERR_INVALID_VALUE,
+                       "Error when loading eBPF filter file");
+        }
+#else
+        SCLogError(SC_ERR_UNIMPLEMENTED, "eBPF support is not build-in");
+#endif
+    }
+
+    if (ConfGetChildValueWithDefault(if_root, if_default, "xdp-filter-file", &ebpf_file) != 1) {
```

Comment:
I'm a bit confused about what things are ifdef'd and which aren't. Seems all references to ebpf should be ifdef'd?